kernels/tiled_mult/tiled.preprocfinal.mlir

```diff
@@ -0,0 +1,85 @@
+#simple_mult_attributes = {
+  indexing_maps = [
+    affine_map<(n) -> (n)>,
+    affine_map<(n) -> (n)>,
+    affine_map<(n) -> (n)>
+  ],
+  iterator_types = ["parallel"],
+  library_call = "snax_hwpe_mult" 
+}
+
+// https://github.com/pulp-platform/hwpe-mac-engine
+// in 'simple_mult' mode, it takes two 32bit fixed-point streams (vectors),
+// A, B and computes D = A * B where '*' is the elementwise product.
+func.func private @snax_is_dm_core() -> i1
+func.func private @snax_is_compute_core() -> i1
+func.func private @debug() -> ()
+func.func public @simple_mult(%A: memref<128xi32, "L3">,
+                             %B: memref<128xi32, "L3">,
+                             %D: memref<128xi32, "L3">) -> () {
+    %is_dm_core = func.call @snax_is_dm_core() : () -> i1
+    %is_compute_core = func.call @snax_is_compute_core () : () -> i1
+    %all_good = arith.constant 0 : i32
+    // This code is run on both cores, note that only the DM core actually runs this, 
+    // and that a barrier is called inside!
+    %A_L1 = "memref.alloc"() <{"alignment" = 64 : i64, "operandSegmentSizes" = array<i32: 0, 0>}> : () -> memref<128xi32, "L1"> 
+    %B_L1 = "memref.alloc"() <{"alignment" = 64 : i64, "operandSegmentSizes" = array<i32: 0, 0>}> : () -> memref<128xi32, "L1">
+    %D_L1 = "memref.alloc"() <{"alignment" = 64 : i64, "operandSegmentSizes" = array<i32: 0, 0>}> : () -> memref<128xi32, "L1">
+    %tile_size = arith.constant 16 : index
+    %c128 = arith.constant 128 : index
+    %c0 = arith.constant 0 : index
+    // Here goes all the code that is run on the DM core
+    "scf.if"(%is_dm_core) ({
+      // Perform the memory transfer on a subview of the memref
+      scf.for %iv = %c0 to %c128 step %tile_size {
+          %tiled_A = "memref.subview"(%A, %iv, %tile_size) <{operandSegmentSizes = array<i32: 1, 1, 1, 0>, static_offsets = array<i64: -9223372036854775808>, static_sizes = array<i64: -9223372036854775808>, static_strides = array<i64: 1>}> : (memref<128xi32, "L3">, index, index) -> memref<?xi32, strided<[1], offset: ?> , "L3">
```

Comment:
What is again the reason for this `-9223372036854775808` magic number?

Reply:
-9223372036854775808 is the smallest possible negative number representable in i64 = -2^63.
memref.subview has variadic inputs for the amount of dynamic offsets, sizes and strides that are given, but at the same time they require a static_sizes attribute:
```
error: 'memref.subview' op requires attribute 'static_sizes'
```
It seems to work for any number (note that they are completely ignored when they are overridden by a dynamic counterpart, but mlir-opt always prints -9223372036854775808, so I thought this was a requirement :smile:

Reply:
Okay so it does start acting weird (it being mlir-opt) if I start playing with sizes, because static negative offsets is still a thing... better  to leave it like this albeit super ugly.